server/service/service_osquery_test.go

```diff
@@ -2458,3 +2468,32 @@ func TestLiveQueriesFailing(t *testing.T) {
 	require.Contains(t, string(logs), "level=error")
 	require.Contains(t, string(logs), "failed to get queries for host")
 }
+
+func TestJitterForHost(t *testing.T) {
+	jh := jitterHashTable{
+		maxCapacity: 1,
+		bucketCount: 30,
+		buckets:     make(map[int]int),
+		cache:       make(map[uint]time.Duration),
+	}
+
+	histogram := make(map[int64]int)
+	for i := 0; i < 1000; i++ {
+		hostID, err := crand.Int(crand.Reader, big.NewInt(10000))
+		require.NoError(t, err)
+		jitter := jh.jitterForHost(uint(hostID.Int64() + 10000))
+		jitterMinutes := int64(jitter.Minutes())
+		histogram[jitterMinutes]++
+	}
+	min, max := math.MaxInt, 0
+	for jitterMinutes, count := range histogram {
+		if count < min {
+			min = count
+		}
+		if count > max {
+			max = count
+		}
+		fmt.Printf("jitterMinutes=%d \t count=%d\n", jitterMinutes, count)
```

Comment:
Seems like the test only makes sense if someone is looking at the output? Would there be a (not too complex) way to check programatically that the histogram has the expected shape/distribution? And we could print the output only if `testing.Verbose()` is true (if we really don't want to use `t.Log` with the potentially annoying prefix messing with the formatting).

Reply:
Unless I'm missing something I believe this bucket approach is predictable enough so that we can check that with `bucketCount=30,maxCapacity=1` e.g. after 30 hosts (with different IDs) check in, all buckets have one item, next 30, all have two..., etc.

Reply:
We can define a variance threshold and make sure it's below that for the count set.